keymapper/injection/global_uinputs.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+# key-mapper - GUI for device specific keyboard mappings
+# Copyright (C) 2021 sezanzeb <proxima@sezanzeb.de>
+#
+# This file is part of key-mapper.
+#
+# key-mapper is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# key-mapper is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with key-mapper.  If not, see <https://www.gnu.org/licenses/>.
+
+
+import evdev
+from evdev.ecodes import EV_KEY
+
+from keymapper.utils import DEV_NAME, is_keyboard_code
+from keymapper.logger import logger
+
+
+class UInput(evdev.UInput):
+    
+    def __init__(self, *args, **kwargs):
+        super().__init__( *args, **kwargs)
+    
+    def can_emit(self, event):
+        """ceck it a event can be emitted by the uinput
+        
+        Wrong events might be injected if the group mappings are wrong
+        """
+        # TODO check for event value especially for EV_ABS
+        try:
+            return event[0] in self.capabilities().keys() and event[1] in self.capabilities()[event[0]]
+        except evdev.uinput.UInputError:
+            logger.debug("uinput for %s is not available", self.name)
+            self.device = self._find_device()
+            return event[0] in self.capabilities().keys() and event[1] in self.capabilities()[event[0]]
```

Comment:
@sezanzeb I get an exception here: `evdev.uinput.UInputError: input device not opened - cannot read capabilities`.

But only when running key-mapper-service directly as a user. As a systemd service everything works as expected.
Is there a way we can set the correct permissions? 

When creating a uinput from the python console it has some extra permissions:
```
$ getfacl /dev/input/event21
getfacl: Entferne führende '/' von absoluten Pfadnamen
# file: dev/input/event21
# owner: root
# group: input
user::rw-
user:jonas:rw-
group::rw-
mask::rw-
other::---
```
compared to the permissions for key-mapper devices:
```
$ getfacl /dev/input/event20
getfacl: Entferne führende '/' von absoluten Pfadnamen
# file: dev/input/event20
# owner: root
# group: input
user::rw-
group::rw-
other::---
```

Reply:
Have you checked the python-evdev code? This happens if `self.device is None`, so I guess `self._find_device()` returns None

Reply:
yes, i did check that code. It is calling `glob.glob(/dev/input/event*)` in order to get all  devices. However, this will only return the devices to which the user has read permissions. A simple fix would be to add the user to the input group. 

But we already know it is possible without adding the user to any group. If we can figure out why the python console creates a UInput with the correct permissions, that would be a better solution.

Reply:
Ok,

the systemd service is supposed to run as root anyway, so I guess it isn't really a problem

Reply:
No, it  doesn't seem to be a problem when running as a service. 
I updated the logic to handle this error in https://github.com/sezanzeb/key-mapper/pull/228/commits/f85ba92826af9c0e9852178031aac7fe0167e80c now we try to find the event* file once, and afterwards we ignore the error and assume the event can be handled by the device.

Reply:
If the service only ever runs with root permission once installed, is this really needed?

Reply:
We can write the events safely. The events get written to /dev/uinput not to /dev/input/event*.

we might not need to send the events anyway if /dev/input/event* is not available, but we should try to find the event* file at least once. 
UInput sleeps for 0.1s before it tries to find the event* file, but apparently that is not always enough. I did observe this on my system.

Reply:
> I did observe this on my system.

but did you run the code with `sudo`?

Reply:
okay, I just did some more testing. This is only a problem when running as a user who is in the input group. Apparently it takes more time to create the event* file if it is crated from a user.

When running as root this is not a problem, so I guess we are fine not catching Error